feat(scheduling): EPI-1179: Basic User Search (User Management Admin Panel)

huytranbes opened the pull request:
### Changes



### Deploys

- [ ] **Deploy to Tamanu Internal** <!-- #deploy -->

### Tests

- [ ] **Run E2E Tests** <!-- #e2e -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

huytranbes committed:
feat(users): Enhance user management view with designations and role names

- Updated user retrieval to include designations and role names in the response.
- Added a new sidebar menu item for user management.
- Refactored UserAdminView to improve layout and user status indicators.
- Enhanced user table to display visibility status, designations, and role names with proper translations.

huytranbes committed:
feat(suggestions): Add suggestions routes and user search functionality

- Introduced new suggestions routes for user-related data.
- Created UserSearchBar component to enhance user search capabilities with advanced fields.
- Updated UserAdminView to integrate UserSearchBar for improved user management experience.
- Refactored buildRoutes to include suggestions routes for better organization.

huytranbes committed:
feat(suggesters): Add 'role' suggester to suggestions service

- Introduced a new suggester for 'role' in the suggestions service.
- Updated SUGGESTER_ENDPOINTS to include 'role' for enhanced suggestion capabilities.

huytranbes committed:
feat(users): Implement advanced user filtering in user retrieval

- Added a new function to create filters for user search based on display name, display ID, email, role, designation, and visibility status.
- Enhanced the user retrieval endpoint to support these filters, improving the search capabilities for user management.
- Included total count for pagination in the response to facilitate better UI handling.

edmofro reviewed (changes requested):
Two tiny nits, but it's great how much you've been able to achieve here in line with the way the existing code is set up :-)

huytranbes committed:
Update packages/central-server/app/admin/users.js

Co-authored-by: Edwin Monk-Fromont <edwin.monkfromont@gmail.com>

huytranbes committed:
prettier format

huytranbes committed:
Merge branch 'feature/epi-1179-basic-user-search-user-management-admin-panel-2' of https://github.com/beyondessential/tamanu into feature/epi-1179-basic-user-search-user-management-admin-panel-2

huytranbes commented on packages/web/app/views/administration/UserAdminView.jsx:
```diff
@@ -116,10 +120,17 @@ const UserTable = React.memo(({ ...props }) => (
+        searchParameters={searchParameters}
+        data-testid="usersearchbar-admin"
+      />
+        <UserTable fetchOptions={searchParameters} data-testid="usertable-mpss" />
```
I have run prettier for this file

edmofro reviewed (approved):
lgtm

huytranbes committed:
Merge branch 'epic-scheduling-enhancements' of https://github.com/beyondessential/tamanu into feature/epi-1179-basic-user-search-user-management-admin-panel-2

huytranbes committed:
feat(users): Enhance user filtering logic in admin panel

- Updated createUserFilters function to accept models as a parameter for improved query capabilities.
- Modified designation filter to utilize a subquery for better performance and accuracy in user retrieval.
